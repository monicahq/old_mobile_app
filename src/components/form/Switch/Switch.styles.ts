import {StyleSheet} from 'react-native';

export const borderColor = '#b5b5b5';

const height = 30;

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    height,
    alignItems: 'center',
  },
  title: {
    flex: 1,
    flexDirection: 'row',
    height,
    alignItems: 'center',
  },
  errorText: {
    color: 'red',
  },
});
