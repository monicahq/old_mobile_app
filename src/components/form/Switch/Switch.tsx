import {primaryColor} from '@theme';
import React, {PureComponent} from 'react';
import {Switch as RNSwitch, SwitchProps, Text, View} from 'react-native';
import {styles} from './Switch.styles';

interface ISwitchProps extends SwitchProps {
  id?: string;
  title: string;
  touched?: boolean;
  error?: string;
}

export class Switch extends PureComponent<ISwitchProps> {
  public static defaultProps = {
    touched: false,
    error: null,
  };

  public render() {
    const {title, error, touched, id, ...props} = this.props;

    return (
      <View style={styles.container}>
        <View key={0} style={styles.title}>
          {!!title && <Text>{title}</Text>}
          <Text style={styles.errorText} testID={`switch-error-${id}`}>
            {' '}
            {!!error && touched && error}
          </Text>
        </View>
        <RNSwitch
          testID={`input-${id}`}
          key={1}
          onTintColor={primaryColor}
          {...props}
        />
      </View>
    );
  }
}
