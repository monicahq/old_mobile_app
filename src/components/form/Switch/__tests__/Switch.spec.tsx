import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {Switch} from '../Switch';

describe('Components', () => {
  describe('Switch', () => {
    it('ios should renders correctly', () => {
      expect(toJson(shallow(<Switch title="My title" />))).toMatchSnapshot();
      expect(
        toJson(shallow(<Switch title="My title" touched={true} />))
      ).toMatchSnapshot();

      expect(
        toJson(
          shallow(<Switch title="My title" touched={true} error="my-error" />)
        )
      ).toMatchSnapshot();

      expect(
        toJson(shallow(<Switch title="My title" error="My errors" />))
      ).toMatchSnapshot();
    });

    it('android should renders correctly', () => {
      jest.mock('Platform', () => ({
        OS: 'android',
      }));

      expect(toJson(shallow(<Switch title="My title" />))).toMatchSnapshot();
      expect(
        toJson(shallow(<Switch title="My title" touched={true} />))
      ).toMatchSnapshot();
    });
  });
});
