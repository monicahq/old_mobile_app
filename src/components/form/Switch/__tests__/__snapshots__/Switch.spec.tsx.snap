// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Switch android should renders correctly 1`] = `
<ForwardRef(View)
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <ForwardRef(View)
    key="0"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <ForwardRef(Text)>
      My title
    </ForwardRef(Text)>
    <ForwardRef(Text)
      style={
        Object {
          "color": "red",
        }
      }
      testID="switch-error-undefined"
    >
       
    </ForwardRef(Text)>
  </ForwardRef(View)>
  <Switch
    key="1"
    onTintColor="#325776"
    testID="input-undefined"
  />
</ForwardRef(View)>
`;

exports[`Components Switch android should renders correctly 2`] = `
<ForwardRef(View)
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <ForwardRef(View)
    key="0"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <ForwardRef(Text)>
      My title
    </ForwardRef(Text)>
    <ForwardRef(Text)
      style={
        Object {
          "color": "red",
        }
      }
      testID="switch-error-undefined"
    >
       
    </ForwardRef(Text)>
  </ForwardRef(View)>
  <Switch
    key="1"
    onTintColor="#325776"
    testID="input-undefined"
  />
</ForwardRef(View)>
`;

exports[`Components Switch ios should renders correctly 1`] = `
<ForwardRef(View)
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <ForwardRef(View)
    key="0"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <ForwardRef(Text)>
      My title
    </ForwardRef(Text)>
    <ForwardRef(Text)
      style={
        Object {
          "color": "red",
        }
      }
      testID="switch-error-undefined"
    >
       
    </ForwardRef(Text)>
  </ForwardRef(View)>
  <Switch
    key="1"
    onTintColor="#325776"
    testID="input-undefined"
  />
</ForwardRef(View)>
`;

exports[`Components Switch ios should renders correctly 2`] = `
<ForwardRef(View)
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <ForwardRef(View)
    key="0"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <ForwardRef(Text)>
      My title
    </ForwardRef(Text)>
    <ForwardRef(Text)
      style={
        Object {
          "color": "red",
        }
      }
      testID="switch-error-undefined"
    >
       
    </ForwardRef(Text)>
  </ForwardRef(View)>
  <Switch
    key="1"
    onTintColor="#325776"
    testID="input-undefined"
  />
</ForwardRef(View)>
`;

exports[`Components Switch ios should renders correctly 3`] = `
<ForwardRef(View)
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <ForwardRef(View)
    key="0"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <ForwardRef(Text)>
      My title
    </ForwardRef(Text)>
    <ForwardRef(Text)
      style={
        Object {
          "color": "red",
        }
      }
      testID="switch-error-undefined"
    >
       
      my-error
    </ForwardRef(Text)>
  </ForwardRef(View)>
  <Switch
    key="1"
    onTintColor="#325776"
    testID="input-undefined"
  />
</ForwardRef(View)>
`;

exports[`Components Switch ios should renders correctly 4`] = `
<ForwardRef(View)
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <ForwardRef(View)
    key="0"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <ForwardRef(Text)>
      My title
    </ForwardRef(Text)>
    <ForwardRef(Text)
      style={
        Object {
          "color": "red",
        }
      }
      testID="switch-error-undefined"
    >
       
    </ForwardRef(Text)>
  </ForwardRef(View)>
  <Switch
    key="1"
    onTintColor="#325776"
    testID="input-undefined"
  />
</ForwardRef(View)>
`;
